test/plog/LogStreamTest.cpp

```diff
@@ -0,0 +1,449 @@
+            bool first_plog = false;
+            for (auto& response:responses){
+                auto& [status, plogId, appended_offset] = response;
+                if (!status.is2xxOK()){
+                    throw std::runtime_error("cannot append data!");
+                }
+                if (!first_plog){
+                    first_plog = true;
+                    _initPlogId = plogId;
+                }
+                K2LOG_D(log::ltest, "{}, {}", plogId, appended_offset);
+            }
+            K2LOG_I(log::ltest, ">>> Test2.1: Write Done");
+            return _logStream->read(_initPlogId, 0, 2000*10005);
+        })
+        // read with continuation
+        .then([this] (auto&& response){
+            uint32_t request_size = 2000*10005;
+            auto& [status, continuation_token, payload] = response;
+            if (!status.is2xxOK()){
+                throw std::runtime_error("cannot read data!");
+            }
+            request_size -= payload.getSize();
+            Payload read_payload;
+            for (auto& b: payload.shareAll().release()) {
+                read_payload.appendBinary(std::move(b));
+            }
+            
+            return seastar::do_with(std::move(read_payload), std::move(request_size), std::move(continuation_token), [&] (auto& read_payload, auto& request_size, auto& continuation_token){
+                return seastar::do_until(
+                    [&] { return request_size == 0; },
+                    [&] {
+                        return _logStream->read(continuation_token, request_size)
+                        .then([&] (auto&& response){
+                            auto& [status, token, payload] = response;
+                            if (!status.is2xxOK()){
+                                throw std::runtime_error("cannot read data!");
+                            }
+                            request_size -= payload.getSize();
+                            continuation_token = std::move(token);
+                            for (auto& b: payload.shareAll().release()) {
+                                read_payload.appendBinary(std::move(b));
+                            }
+                            return seastar::make_ready_future<>();
+                        });
+                    }
+                )
+                .then([&] (){
+                    return seastar::make_ready_future<Payload>(std::move(read_payload));
+                });
+            });
+        })
+        // write another 1000 Strings to the same log stream, these writes will trigger a plog switch as well
+        .then([this, header] (auto&& payload){
+            K2LOG_I(log::ltest, ">>> Test2.2: Read Done");
+            String str;
+            uint32_t count = 0;
+            payload.seek(0);
+            while (payload.getDataRemaining() > 0){
+                payload.read(str);
+                K2EXPECT(log::ltest, str, header);
```

Comment:
validate str contents, not just check if it is same as header.

Reply:
I think this expect will check the value of str equal to the value of the header. Is it correct?